src/puppetlabs/puppetdb/utils.clj

```diff
@@ -644,3 +643,18 @@
 
 (defn schedule-with-fixed-delay [s f initial-delay delay]
   (.scheduleWithFixedDelay s f initial-delay delay TimeUnit/MILLISECONDS))
+
+(defn pprint-json-parse-exception
+  "Returns a parsed JsonParseException message"
+  [error query]
+  (let [lines (str/split-lines (.getMessage error))
+        positions (re-find #"line: (\d+), column: (\d+)" (lines 1))
```

Comment:
I noticed this regex was getting the index values you wanted and also picking up the `"line: 1 column: 5"` part of the string. I saw it getting the following array in the `reports_test.clj/query-report-with-malformed-json` test: `["line: 1, column: 5" "1" "5"]`. If it's easy to restrict the regex to get only the index values we might want that. 

We could also use destructuring here in the `let` block to pull out the `line-index`and `column-index` values from the `re-find`. Right now with it would look something like this: 
`[_ line-index column-index] (re-find #"line: (\d+), column: (\d+)" (second lines))`
but we could drop the throw away `_` binding there if we restrict the regex. We'd still need to convert the strings to ints later on though, so the destructuring might not be ideal here unless we can make sure we'll only ever get the index values back from the regex and convert them to ints before binding them.

Reply:
I don't think there's a way to return the capturing groups without the full match